src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersions.java

```diff
@@ -127,6 +128,54 @@ public boolean containsAnyOf(ArtifactVersions versions) {
     return artifactVersions.get().sortByReleaseDate();
   }
 
+  /**
+   * Filter {@link ArtifactVersions} by major version.
+   *
+   * @param version The artifact version.
+   * @return Filtered {@link ArtifactVersions}.
+   */
+  public ArtifactVersions filterArtifactsByMajorVersion(String version) {
+    Optional<SemanticVersion> currentVersion = SemanticVersion.parse(version);
+    if (!currentVersion.isPresent()) {
+      return ArtifactVersions.ofNothing();
+    }
+
+    String currentMajorVersion = buildMajorVersion(currentVersion.get().getMajor());
+    Optional<String> nextMajorVersion = nextMajorVersionOf(version);
+
+    return new ArtifactVersions(elements.stream().filter(v -> v.hasValidSemanticVersion())
+        .filter(v -> new ComparableVersion(v.version())
+            .compareTo(new ComparableVersion(currentMajorVersion)) >= 0
+            && new ComparableVersion(v.version())
+                .compareTo(new ComparableVersion(nextMajorVersion.get())) < 0)
```

Comment:
Could it be simplified a bit?

```suggestion
    return new ArtifactVersions(
        elements.stream()
            .filter(v::hasValidSemanticVersion)
            .map(v -> new ComparableVersion(v.version()))
            .filter(v -> v.compareTo(currentMajorVersion) >= 0 && v.compareTo(nextMajorVersion) < 0)
```